File: javascript/fizzbuzz.js
var fizzbuzz = function(x) {
  if ( x > 1 ) {
    fizzbuzz( x - 1 );
  }
  if ( (x % 3 === 0) && (x % 5 === 0) ) {
    console.log( "fizzbuzz" );
  } else if ( x % 3 === 0 ) {
    console.log( "fizz" );
  } else if ( x % 5 === 0 ) {
    console.log( "buzz" );
  } else {
    console.log( x );
  }
}

fizzbuzz( 100 );

File: clojure/fizzbuzz.clj
; fizzbuzz in Clojure
(defn fizzbuzz 
  ([] (fizzbuzz 100))
  ([n] 
  (map 
    #(if (= (rem % 15) 0) 
      "fizzbuzz" 
      (if (= (rem % 3) 0) 
        "fizz" 
        (if (= (rem % 5) 0) 
          "buzz" 
          %)))
  (range 1 n))))

(fizzbuzz)
(fizzbuzz 100)
